docs/CodingGuidelines.md

```diff
@@ -0,0 +1,174 @@
+1. React components will be exported within https://github.com/awslabs/iot-app-kit/tree/main/packages/react-components with no prefix, i.e. LineChart
+1. Components can also be created in other repositories, but must ultimately be exported from either @iot-app-kit/components and @iot-app-kit/react-components (or both)
+1. All components prior to release must have public facing documentation within https://github.com/awslabs/iot-app-kit/tree/main/docs, including updating the https://github.com/awslabs/iot-app-kit with the relevant information.
+1. Components must work across the commonly used browsers. Any browser that has over 1% of total internet usage as defined by https://caniuse.com/usage-table is considered a commonly used browser. (which does not include IE11.
+
+1. All IoT App Kit components contain a `size` property, which can be "XS", "S", "M", "L", "XL", or "XXL".
+
+   These sizes do not affect the overall width and height of the element, but instead form a guide for sizing widget elements such as font size, line thickness, symbol size, and level of detail. Each size has a standard sizing for various elements, as well as a dimension which the widget should look good at. For example, an XS widget should be displayable at a size of 75px by 75px, but a user could display a XS widget in a larger container if they desired to. However a XXL widget if put in a small container, will not work well.
+
+   | Size | Primary font size | Large font size |Icon size | Looks good at dimension |
+   |------|-------------------|-----------------|----------| ------------------------|
+   | XS    | 14                     | 20                   | 16       | 75px by 75px|
+   | S     | 16                       | 24                     | 20       | 100px by 100px|
+   | M     | 20                     | 32              | 32       | 150px by 150px|
+   | L     | 24                      | 48              | 48       | 200px by 200px|
+   | XL    | 32                | 60              | 60       | 300px by 300px|
+   | XXL   | 48                | 96             | 96        | 500px by 500px|
+
+   (Note: this specification is not exhaustive and subject to change. Size-affected elements will vary by component
+
+1. Components must support internationalization via messageOverrides (i.e. avoid hard coded phrases)
+
+   for example,
+   ```
+   <iot-line-chart  
+     messageOverrides={{  
+       loadingLabel: 'Loading...',  
+       invalidDataType: 'your data is not a supported type',  
+     }}  
+     ...  
+   />  
+   ```
+
+1. Components should support various data sources, via either a query or queries property.
+   for example,
+     ```
+      <iot-line-chart  
+      queries={[query.timeSeriesData(...)]}  
+      ...  
+    />  
+    ```
+1. It is ok to have data-source specific features. Not all data-source features will be generic in nature, but that shouldn’t prevent us from exposing them to make it easy for customers to utilize the feature
```

Comment:
nit: missing period

In some of the bullet points below there are commas instead of periods. Numbering looks off here too.

Reply:
utilizing best practice for numbering here: https://google.github.io/styleguide/docguide/style.html#use-lazy-numbering-for-long-lists

Reply:
fixed missing period